go.mod

```diff
@@ -51,7 +51,7 @@ require (
 	github.com/libp2p/go-doh-resolver v0.4.0
 	github.com/libp2p/go-libp2p v0.36.2
 	github.com/libp2p/go-libp2p-http v0.5.0
-	github.com/libp2p/go-libp2p-kad-dht v0.25.2
+	github.com/libp2p/go-libp2p-kad-dht v0.26.0
```

Comment:
v0.26.0 was missing some commits, but testing with https://github.com/libp2p/go-libp2p-kad-dht/pull/980 also fails in the same place ([log](https://github.com/ipfs/kubo/actions/runs/10493632540/job/29068142778#step:9:161)).

Needs analysis. Kubo being able to resolve thing when it is not expected to is a weird regression, I suspect kad-dht client got better somehow, and helia-interop test no longer  making sense, may be requiring update.

Reply:
Uppon inspection it feels the failing "helia→pubsub→kubo" test is a bit hacky and does not follow what is done is his simbling  "kubo→pubsub→helia":

https://github.com/ipfs/helia/blob/777d868c2c27e3ef0eaafab5c255a4f60e275874/packages/interop/src/ipns-pubsub.spec.ts#L81-L88:

```js
// first publish should fail because kubo isn't subscribed to key update channel
await expect(name.publish(peerId, cid)).to.eventually.be.rejected()
  .with.property('message', 'PublishError.NoPeersSubscribedToTopic')

// should fail to resolve the first time as kubo was not subscribed to the pubsub channel
await expect(last(kubo.api.name.resolve(peerId, {
  timeout: 100
}))).to.eventually.be.undefined()
```

This is brittle, it calls `name.publish`  when it should not – there are better ways  of confirming that peers are or are not subscribed to a topic.

I'll provide upstream fix + switch to it before merging this.